dds/DCPS/DiscoveryBase.h

```diff
@@ -1051,170 +1120,68 @@ namespace OpenDDS {
+              //return;
             }
-
-            if (call_writer) {
-              RepoId reader_participant = reader;
-              reader_participant.entityId = ENTITYID_PARTICIPANT;
-              DatawriterCryptoHandleMap::const_iterator iter =
-                local_writer_crypto_handles_.find(writer);
-
-              // It might not exist due to security attributes, and that's OK
-              if (iter != local_writer_crypto_handles_.end()) {
-                DDS::Security::DatawriterCryptoHandle dwch = iter->second;
-                DDS::Security::DatareaderCryptoHandle drch =
-                  generate_remote_matched_reader_crypto_handle(
-                    reader_participant, dwch, relay_only_readers_.count(reader));
-                remote_reader_crypto_handles_[reader] = drch;
-                DatareaderCryptoTokenSeqMap::iterator t_iter =
-                  pending_remote_reader_crypto_tokens_.find(reader);
-                if (t_iter != pending_remote_reader_crypto_tokens_.end()) {
-                  DDS::Security::SecurityException se;
-                  if (!keyexg->set_remote_datareader_crypto_tokens(iter->second, drch, t_iter->second, se)) {
-                    ACE_ERROR((LM_ERROR, ACE_TEXT("(%P|%t) ERROR: ")
-                      ACE_TEXT("(%P|%t) ERROR: DiscoveryBase::match() - ")
-                      ACE_TEXT("Unable to set pending remote datareader crypto tokens with crypto ")
-                      ACE_TEXT("key exchange plugin. Security Exception[%d.%d]: %C\n"),
-                        se.code, se.minor_code, se.message.in()));
-                  }
-                  pending_remote_reader_crypto_tokens_.erase(t_iter);
-                }
-                EndpointSecurityAttributesMap::const_iterator s_iter =
-                  local_writer_security_attribs_.find(writer);
-                if (s_iter != local_writer_security_attribs_.end() &&
-                    (s_iter->second.is_submessage_protected || s_iter->second.is_payload_protected)) {
-                  create_and_send_datawriter_crypto_tokens(dwch, writer, drch, reader);
-                }
+          } else if (!reader_local) {
+            if (type_lookup_service_ && !type_lookup_service_->type_object_in_cache(reader_type_info->minimal.typeid_with_size.type_id)) {
+              XTypes::TypeIdentifierSeq type_ids;
+              type_ids.append(reader_type_info->minimal.typeid_with_size.type_id);
+              md.rpc_sequence_number = ++(type_lookup_service_->rpc_sequence_number_);
+              MatchingDataIter md_it = matching_data_buffer_.find(MatchingPair(writer, reader));
+              if (md_it != matching_data_buffer_.end()) {
+                // TLS_TODO: is this scenario possible?
+                md_it->second = md;
+              } else {
+                matching_data_buffer_.insert(std::make_pair(MatchingPair(writer, reader), md));
               }
+              //send_type_lookup_request(type_ids, reader);
```

Comment:
Why is this commented-out?

Reply:
After uncommenting many Jenkins tests fail:
https://prb.objectcomputing.com/jenkins/job/OpenDDS_Pull_Requests/2834/.
I re-ran some of them locally and they pass. Not sure yet if it's due to some Type Identifiers not being added to the map as they should, or because the comparison operator does not work.